task/gcp/task.go

```diff
@@ -352,8 +352,12 @@ func (t *Task) Events(ctx context.Context) []common.Event {
 	return t.Attributes.Events
 }
 
-func (t *Task) Status(ctx context.Context) common.Status {
-	return t.Attributes.Status
+func (t *Task) Status(ctx context.Context) (common.Status, error) {
```

Comment:
This function is repeated 3 times 😅

Reply:
I think it could go in the generic task?

Reply:
Not quite, because of the same architectural issue on https://github.com/iterative/terraform-provider-iterative/pull/237#discussion_r750507783. If you find a (subjectively) more elegant way, feel free to follow up.